fdbserver/ptxn/test/RealTLog.actor.cpp

```diff
@@ -195,13 +197,14 @@ ACTOR Future<Void> TLogDriverContext::peekCommitMessages_impl(std::shared_ptr<Te
 		rd.readBytes(tagCount * sizeof(Tag));
 
 		// deserialize span id
-		SpanContextMessage contextMessage;
-		rd >> contextMessage;
-
-		// deserialize mutation header
-		rd >> messageLength >> sub >> tagCount;
-		rd.readBytes(tagCount * sizeof(Tag));
+		if (FLOW_KNOBS->WRITE_TRACING_ENABLED) {
+			SpanContextMessage contextMessage;
+			rd >> contextMessage;
```

Comment:
It seems this should be deserialized after line 206. See https://github.com/jzhou77/foundationdb/blob/f0da524314a8efca9c3c162a2e6edf82b668b448/fdbserver/LogSystem.h#L1115-L1119

Reply:
are you suggesting to deserialize the spanID _after_ the mutations (by moving this after line 206)? (sorry if I misunderstand). 

that does not seem right... the spanID is serialized before the mutations; see the call to `writeTransactionInfo` [here](https://github.com/jzhou77/foundationdb/blob/f0da524314a8efca9c3c162a2e6edf82b668b448/fdbserver/LogSystem.h#L1040 ). 

I am building the request using `writeTypedMessage` which first serializes the spanID (plus some other stuff), then serializes the mutations..

Reply:
Yes, I think `SpanContextMessage` should be deserialized after the metadata. This is because this message is treated as a special `MutationRef` in `writeTransactionInfo()`, where we first write size, subseq, tags, and then this message. As you can see, the `SpanContextMessage` is the last one written.

```
		wr << uint32_t(0) << subseq << uint16_t(prev_tags.size());
		for (auto& tag : prev_tags)
			wr << tag;
		wr << contextMessage;
```

Reply:
I see you mean the span index should be deserialized after the `subseq` and `tags`  (you did not means the mutations). We are on the same page.